### DISCUSSION TITLE
ScriptReference/Font.GetPathsToOSFonts

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#605] [Notes]
# ScriptReference/Font.GetPathsToOSFonts

https://docs.unity3d.com/ScriptReference/Font.GetPathsToOSFonts.html

**Comment by radiatoryang** [comment]
On Android, the paths could look like "/system/fonts/NotoSansCJK-Regular.ttc"
On Windows, the paths could look like "C:/Windows/Fonts/segoeui.ttf" or sometimes like "C:/Windows/Fonts/ARIALN.TTF"

For searching, you'll likely only care about the font name... see https://docs.unity3d.com/ScriptReference/Font.GetOSInstalledFontNames.html
but make sure you still pass the full path into the Font constructor!

see https://forum.unity.com/threads/load-font-from-mobile-device-and-assign-as-fallback-tmp_fontasset.657151/ for history and usage

Windows type faces: https://en.wikipedia.org/wiki/List_of_typefaces_included_with_Microsoft_Windows

example Android type face list for CJK:
"Roboto",
"Noto Sans CJK SC",
"Noto Sans Mono CJK SC",
"Noto Sans SC",
"Noto Sans CJK TC",
"Noto Sans Mono CJK TC",
"Noto Sans TC",
"Noto Sans CJK KR",
"Noto Sans Mono CJK KR",
"Noto Sans KR",
"Droid Sans Hangul",
"Noto Sans CJK JP",
"Noto Sans Mono CJK JP",
"Noto Sans JP",
"Droid Sans Japanese",
"MotoyaLMaru",
"NanumGothic",
"Droid Sans",
"Droid Sans Fallback",
"DroidSansFallback"
